src/Akka.Streams.Kafka/Stages/Consumers/Committers.cs

```diff
@@ -12,24 +20,45 @@ internal interface IInternalCommitter
         /// <summary>
         /// Commit all offsets (of different topics) belonging to the same stage
         /// </summary>
-        Task Commit();
+        Task Commit(ImmutableList<PartitionOffset> offsets);
     }
-
+    
     /// <summary>
-    /// This is a simple committer using kafka consumer directly (not using consumer actor, etc)
+    /// Used by <see cref="CommittableSourceMessageBuilder{K,V}"/> to commit messages by
+    /// sending <see cref="KafkaConsumerActorMetadata.Internal.Commit"/> to <see cref="KafkaConsumerActor{K,V}"/>
     /// </summary>
-    internal class KafkaCommitter<K, V> : IInternalCommitter
+    internal class KafkaAsyncConsumerCommitter : IInternalCommitter
     {
-        private readonly IConsumer<K, V> _consumer;
+        private readonly TimeSpan _commitTimeout;
+        private readonly Lazy<IActorRef> _consumerActor;
 
-        public KafkaCommitter(IConsumer<K, V> consumer)
+        public KafkaAsyncConsumerCommitter(Func<IActorRef> consumerActorFactory, TimeSpan commitTimeout)
         {
-            _consumer = consumer;
+            _commitTimeout = commitTimeout;
+            _consumerActor = new Lazy<IActorRef>(consumerActorFactory);
```

Comment:
Need deferred execution here because consumer actor is not instantiated at the moment of `KafkaAsyncConsumerCommitter` creation

Reply:
Makes sense